demo_app/app/backend/directory_helper.js

```diff
@@ -8,20 +8,24 @@ export let computeDirectorySize = function(localPath) {
   let size = 0;
   let stat;
   let tempPath;
-  let contents = fs.readdirSync(localPath);
-  for (var i in contents) {
-    tempPath = localPath + '/' + contents[i];
-    stat = fs.statSync(tempPath);
-    if (stat.isDirectory()) {
-      size += computeDirectorySize(tempPath);
-    } else {
-      if (env.isFileUploadSizeRestricted && stat.size > env.maxFileUploadSize) {
-        throw new Error('File more than ' + (env.maxFileUploadSize / 1000000) + ' Mb can not be uploaded');
+  try {
+    let contents = fs.readdirSync(localPath);
```

Comment:
I wonder why the use of Sync calls here and not async, is there a performance hit here?